addons/redis/templates/componentdefinition-redis.yaml

```diff
@@ -0,0 +1,245 @@
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: ComponentDefinition
+metadata:
+  name: {{ include "redis.componentDefName" . }}
+  labels:
+    {{- include "redis.labels" . | nindent 4 }}
+spec:
+  provider: kubeblocks
+  description: A Redis v7.0 component definition for Kubernetes
+  serviceKind: redis
+  serviceVersion: 7.0.6
+  services:
+    - name: redis
+      serviceName: redis
+      spec:
+        ports:
+          - name: redis
+            port: 6379
+            targetPort: redis
+      roleSelector: primary
+  updateStrategy: BestEffortParallel
+  volumes:
+    - name: data
+      needSnapshot: true
+  roles:
+    - name: primary
+      serviceable: true
+      writable: true
+    - name: secondary
+      serviceable: false
+      writable: false
+  roleArbitrator: Lorry
+  logConfigs:
+    {{- range $name,$pattern := .Values.logConfigs }}
+    - name: {{ $name }}
+      filePathPattern: {{ $pattern }}
+    {{- end }}
+  monitor:
+    builtIn: false
+    exporterConfig:
+      scrapePort: {{ .Values.metrics.service.port }}
+      scrapePath: "/metrics"
+  configs:
+    - name: redis-replication-config
+      templateRef: redis7-config-template
+      constraintRef: redis7-config-constraints
+      namespace: {{ .Release.Namespace }}
+      volumeName: redis-config
+    - name: redis-metrics-config
+      templateRef: redis-metrics-config
+      namespace: {{ .Release.Namespace }}
+      volumeName: redis-metrics-config
+      defaultMode: 0444
+  scripts:
+    - name: redis-scripts
+      templateRef: redis-scripts
+      namespace: {{ .Release.Namespace }}
+      volumeName: scripts
+      defaultMode: 0555
+  systemAccounts:
+    ## TODO: the other accounts should be created for backward compatibility
+    - name: default
+      initAccount: true
+      passwordGenerationPolicy:
+        length: 10
+        numDigits: 5
+        numSymbols: 0
+        letterCase: MixedCases
+  vars:
+    - name: REDIS_DEFAULT_USER
+      valueFrom:
+        credentialVarRef:
+          ## reference the current component definition name
+          compDef: {{ include "redis.componentDefName" . }}
+          name: default
+          optional: false
```

Comment:
> since required is the default option, you do not specify it explicitly

This should be indicated in the API documentation as the default value, which is currently missing. I will add it.